### PR TITLE
cluster-api-provider-docker: add presubmit job and OWNERS file

### PR DESCRIPTION
**Pull request by neolit123**
also include:
- milestone plugin for this repo
- testgrid entries

once the owners and yamls are LGTM'd we can assign a test-infra approver.

/assign @chuckha 
/kind feature
/area config
/sig cluster-lifecycle

### CODE REVIEW

**Comment by detiber** [comment]
doh, forgot that the test-grid config is required.

**Comment by neolit123** [comment]
hm, shouldn't be required unless these two options force it:
```
    always_run: true
    optional: false
```

https://github.com/kubernetes/test-infra/pull/12211/files

**Comment by chuckha** [comment]
/lgtm 

! thank you for getting this set up

File: prow/plugins.yaml [hunk 1]
```diff
@@ -604,7 +604,7 @@ plugins:
 
   kubernetes-sigs/gcp-compute-persistent-disk-csi-driver:
   - release-note
-  
+
```
Comment (neolit123) [inline prow/plugins.yaml]: auto-whitespace cleanup by my editor let me know if needs revert.

**Comment by neolit123** [comment]
/assign @fejta 
PTAL at this addition of a presubmit job for cluster-api-provider-docker.
seems that pull-test-infra-bazel  is stuck in infinite loop?

**Comment by neolit123** [comment]
/unassign @fejta 

sorry didn't see:
`/assign @spiffxp`

**Comment by stevekuznetsov** [comment]
/test pull-test-infra-bazel

It might be hitting the scheduling bug

**Comment by neolit123** [comment]
tests are passing here.

**Pull request merged**